blogposts/announcing-sourcegraph-3.1.md

```diff
@@ -0,0 +1,121 @@
+
+<!-- (ryan): Hyperlink to headings in above paragraph -->
+
+What is Sourcegraph? Sourcegraph is an open-source, self-hosted, cross-repository code search and navigation tool, with an efficient web interface and "feels-like-native" integration into your code host. [Install or upgrade Sourcegraph.](#install-or-upgrade)
+
+## Find answers in code more easily with the new query builder
+
+<!-- TODO (ryan): Below paragraph needs work as it sounds awkward -->
+
+Did you know that in addition to code, you can search commit diffs, commit messages and symbols?
+
+Even experienced Sourcegraph users will find something new. Give it a try and let us know what you think via [GitHub](https://github.com/sourcegraph/sourcegraph/issues/new/choose) or [Twitter](https://twitter.com/srcgraph).
+
+<!-- TODO (ryan): GIF using the query builder -->
+
+## Improved go-to-definition and hover tooltips for all languages
+
+We've improved Sourcegraph's [code intelligence extensions](https://sourcegraph.com/extensions?query=category%3A%22Programming+languages%22), which let you go to definitions, find references, and see hover tooltips on your code (in all languages).
+
+- The docstrings in hover tooltips are more accurate.
+- Indented sections in docstrings are rendered as code blocks in hovers.
+- Tokens in comments ignored unless they look like code (e.g., if they're wrapped in backticks or look like a function call `foo()`)
+
+We're actively improving code intelligence in all languages, so please [send us feedback](https://github.com/sourcegraph/sourcegraph/issues/new?assignees=&labels=&template=feature_request.md&title=Language%20extension%20improvement%20for)!```
+
+## Site admin improvements
+
+We've added a couple features to help site admins manage a Sourcegraph instance in an enterprise environment:
+
+- Improved [management console](https://docs.sourcegraph.com/admin/management_console) JSON editor for critical configuration, now with editor completion, inline documentation, and validation.
+- Querying users by email address.
+- [Prevent users from changing their username](https://docs.sourcegraph.com/admin/config/critical_config#reference) (from the username set by SSO)
+
+Making configuration changes through the JSON editor is now easier too with easy to access documentation in the editor itself.
+
+<!-- TODO (ryan): GIF showing editor using the help tool tip with an item that requires a choice of values -->
+
+## Improved configuration management and documentation
+
+Sourcegraph 3.0 introduced the [management console](https://about.sourcegraph.com/blog/sourcegraph-3.0#management-console-and-site-config-improvements) for editing critical configuration to allow recovery from misconfiguration in the site admin.
+
+We've [updated our documentation](https://docs.sourcegraph.com/admin/config) to reflect this change, separating configuration into two sections:
```

Comment:
This bullet list seems like it could be deleted. The following paragraph has both links and describes what changed.

Reply:
Done!